apps/channels/models.py

```diff
@@ -33,21 +34,26 @@ class ChannelPlatform(models.TextChoices):
     SLACK = "slack", "Slack"
 
     @classmethod
-    def for_dropdown(cls, team):
-        """Returns availabel platforms for this team. Platforms available through messaging providers will only be
-        returned if a provider is configured.
-        """
+    def for_dropdown(cls, used_platforms, team) -> OrderedDict:
+        """Returns a dictionary of available platforms for this team. Available platforms will have a `True` value"""
         from apps.service_providers.models import MessagingProvider
 
-        available_platforms = [cls.TELEGRAM]
-        if settings.SLACK_ENABLED:
-            available_platforms.append(cls.SLACK)
+        all_platforms = cls.as_list(exclude=[cls.API, cls.WEB])
+        platform_availability = OrderedDict.fromkeys(all_platforms, value=False)
```

Comment:
no need to use `OrderedDict`, you can use `dict.fromkeys`. Dictionaries are always ordered.

Reply:
I see you changed this in a later commit